x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/ranges.tsx

```diff
@@ -161,15 +134,44 @@ export const rangeOperation: OperationDefinition<RangeIndexPatternColumn, 'field
       sourceField: field.name,
     };
   },
-  toEsAggsConfig: (column, columnId) => {
-    const params = getEsAggsParams(column);
-    return {
+  toEsAggsFn: (column, columnId) => {
+    const { sourceField, params } = column;
+    if (params.type === MODES.Range) {
+      return buildExpressionFunction<AggFunctionsMapping['aggRange']>('aggRange', {
+        id: columnId,
+        enabled: true,
+        schema: 'segment',
+        field: sourceField,
+        ranges: JSON.stringify(
+          params.ranges.filter(isValidRange).map<Partial<RangeType>>((range) => {
+            if (isFullRange(range)) {
+              return range;
+            }
+            const partialRange: Partial<RangeType> = { label: range.label };
+            // be careful with the fields to set on partial ranges
+            if (isValidNumber(range.from)) {
+              partialRange.from = range.from;
+            }
+            if (isValidNumber(range.to)) {
+              partialRange.to = range.to;
+            }
+            return partialRange;
+          })
+        ),
+      }).toAst();
+    }
+    return buildExpressionFunction<AggFunctionsMapping['aggHistogram']>('aggHistogram', {
       id: columnId,
       enabled: true,
-      type: column.params.type,
       schema: 'segment',
-      params,
-    };
+      field: sourceField,
+      // fallback to 0 in case of empty string
+      maxBars: params.maxBars === AUTO_BARS ? undefined : params.maxBars,
+      interval: 'auto',
+      has_extended_bounds: false,
+      min_doc_count: true,
```

Comment:
Important: Please confirm that the update I made to `min_doc_count` doesn't have any unexpected effects on Lens, because I think it might actually be changing the behavior...

According to the histogram agg type, `min_doc_count` should be a `boolean` (not a number). It writes out the param as follows:
```ts
         if (aggConfig.params.min_doc_count) {
            output.params.min_doc_count = 0;
          } else {
            output.params.min_doc_count = 1;
          }
```

So as you can see, setting it to `true` will return a value of `0`, which I why I changed it to `true` here as it was previously set to `0`.

_However_, since the check is just confirming whether `min_doc_count` is truthy, `0` might instead cause it to actually be set to `1` (super confusing). So it's possible that lens was inadvertently relying on this behavior unknowingly, which is why I'd appreciate a second set of eyes on this one.

Reply:
Good point, yes we are relying on this behavior, could you flip the boolean?

Otherwise histogram is hard to use on a table:

<img width="1465" alt="Screenshot 2020-12-08 at 14 03 28" src="https://user-images.githubusercontent.com/1508364/101487273-3727db00-395e-11eb-924c-9f0ce90bf100.png">

<img width="855" alt="Screenshot 2020-12-08 at 14 03 34" src="https://user-images.githubusercontent.com/1508364/101487285-3a22cb80-395e-11eb-9565-de279d6c042d.png">